term/src/terminalstate.rs

```diff
@@ -1977,6 +2007,13 @@ impl TerminalState {
                 self.insert = false;
             }
 
+            Mode::SetMode(TerminalMode::Code(TerminalModeCode::AutomaticNewline)) => {
```

Comment:
would you mind adding a unit test for this?
It can be similar to the one from your previous PR!